cli/generators/crud/templates/components/addPage.js

```diff
@@ -0,0 +1,72 @@
+import React, { PropTypes } from 'react';
+import { browserHistory } from 'react-router';
+import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import toastr from 'toastr';
+import * as <%= name %>Actions from '../../actions/<%= name %>Actions';
+import <%= ucName %>Form from './<%= ucName %>Form';
+
+class <%= ucName %>AddPage extends React.Component {
+  constructor(props, context) {
+    super(props, context);
+
+    this.state = {
+      <%= name %>: Object.assign({}, props.<%= name %>),
+      saving: false
+    };
+
+    this.handleSave = this.handleSave.bind(this);
+  }
+
+  handleSave(<%= name %>) {
+    this.setState({ saving: true });
+    this.props.actions.create<%= ucName %>(<%= name %>)
+      .then(() => {
+        this.setState({ saving: false });
+        toastr.success('<%= ucName %> created successfully');
+        browserHistory.push('/app/<%= pluralizedName %>');
+      })
+      .catch(error => {
+        this.setState({ saving: false });
```

Comment:
I'm fine either way. My approach duplicates less code but it's not intuitive at all (at least not for me), so now I'm regretting a little bit that I suggested it. If it wasn't obvious for us while discussing it, very likely it not going to be useful for whoever works with this code, so if we do that, let's leave a comment indicating that it will work like a "finally".